packages/runtime/src/runtime/navigateApi.tsx

```diff
@@ -1,26 +1,41 @@
-import type {
-  NavigateScreenAction,
-  ScreenContextDefinition,
+import {
+  evaluate,
+  findExpressions,
+  type ScreenContextDefinition,
+  type EnsembleStorage,
 } from "@ensembleui/react-framework";
-import { isString } from "lodash-es";
+import { cloneDeep, isString, set } from "lodash-es";
 import type { NavigateFunction } from "react-router-dom";
 
 export const navigateApi = (
-  targetScreen: NavigateScreenAction,
+  targetScreen: string,
+  data: unknown,
   screenContext: ScreenContextDefinition,
+  storage: EnsembleStorage,
   navigate: NavigateFunction,
+  customScope?: Record<string, unknown>,
 ): void => {
-  const hasOptions = !isString(targetScreen);
-  const screenName = hasOptions ? targetScreen.name : targetScreen;
-
   // find the matching screen
   const matchingScreen = screenContext?.app?.screens.find(
-    (s) => s.name.toLowerCase() === screenName.toLowerCase(),
+    (s) => s.name.toLowerCase() === targetScreen.toLowerCase(),
   );
 
   if (!matchingScreen) {
     return;
   }
 
-  navigate(`/${matchingScreen.name.toLowerCase()}`);
+  // set additional inputs
+  const inputs = isString(targetScreen) && data ? cloneDeep(data) : {};
+
+  const expressionMap: string[][] = [];
+  findExpressions(inputs, [], expressionMap);
+  expressionMap.forEach(([path, value]) => {
+    const result = evaluate(screenContext, value, {
+      ensemble: { storage },
+      ...customScope,
+    });
+    set(inputs, path, result);
+  });
+
+  navigate(`/${matchingScreen.name.toLowerCase()}`, { state: inputs });
```

Comment:
You don't need to do the evaluation, just update the function signature and pass the state to `navigate`. The rest is taken care of by the `evaluate` fn